modules/src/ics24_host/path.rs

```diff
@@ -0,0 +1,86 @@
+/// Path-space as listed in ICS-024
+/// https://github.com/cosmos/ics/tree/master/spec/ics-024-host-requirements#path-space
+/// Some of these are implemented in other ICSs, but ICS-024 has a nice summary table.
+use crate::ics24_host::identifier::{ChannelId, ClientId, ConnectionId, PortId};
+use std::fmt::{Display, Formatter, Result};
+
+/// IBC Query Path is hard-coded
+pub const IBC_QUERY_PATH: &str = "store/ibc/key";
+
+/// The Path enum abstracts out the different sub-paths
+pub enum Path {
+    ClientType(ClientId),
+    ClientState(ClientId),
+    ConsensusState(ClientId, u64),
+    ClientConnections(ClientId),
+    Connections(ConnectionId),
+    Ports(PortId),
+    ChannelEnds(PortId, ChannelId),
+    SeqSends(PortId, ChannelId),
+    SeqRecvs(PortId, ChannelId),
+    SeqAcks(PortId, ChannelId),
+    Commitments(PortId, ChannelId, u64),
+    Acks(PortId, ChannelId, u64),
+}
+
+impl Path {
+    /// Indication if the path is provable.
+    pub fn is_provable(&self) -> bool {
+        match &self {
+            Path::ClientState(_) => false,
+            Path::ClientConnections(_) => false,
+            Path::Ports(_) => false,
+            _ => true,
+        }
+    }
+
+    /// into_bytes implementation
+    pub fn into_bytes(self) -> Vec<u8> {
+        self.to_string().into_bytes()
+    }
+}
+
+/// The Display trait adds the `.to_string()` method to the Path struct
+/// This is where the different path strings are constructed
+impl Display for Path {
```

Comment:
We could perhaps consider using the [`derive_more`](https://docs.rs/derive_more) crate to make this more readable. Downside is that it pulls in quite a few dependencies, though they are compile-time only. This can be mitigated a bit by only using the [features](https://github.com/JelteF/derive_more/blob/ece3773ef924e5d2691a61797d4d40fdfbfff56b/Cargo.toml#L41-L65) we need.

Reply:
I'll look into this in a new issue. Compile-time dependencies are not necessarily good, because Ethan is already complaining about the compile time, but readability is always welcome!